CheckerUI_xforms/CheckerUI/CheckerUI/CheckerUI/Helpers/Order/OrderItemView.cs

```diff
@@ -1,248 +1,134 @@
-                FontFamily = "FAS",
-                Command = StartCommand
-            };
-            
-            m_StopButton = new Button
-            {
-                Text = FontAwesomeIcons.Stop,
-                HorizontalOptions = LayoutOptions.StartAndExpand, VerticalOptions = LayoutOptions.StartAndExpand,
-                BackgroundColor = Color.Transparent,
-                FontFamily = "FAS",
-                Command = StopCommand
-            };
-            GenerateGrid();
+
         }
-        
-        public void GenerateGrid()
+
+        public OrderItemView(OrderItemModel i_item)
         {
-            int i = 0;
-            foreach (var label in m_Labels)
-            {
-                m_Grid.Children.Add(label, 0, 3, i, i+1);
-                i++;
-            }
-            m_Grid.Children.Add(m_DoneButton, 0, 1, 4, 5);
-            m_Grid.Children.Add(m_StartButton, 1, 2, 4, 5);
-            m_Grid.Children.Add(m_StopButton, 2, 3, 4, 5);
+            feelColorsState();
+            m_orderItem = new OrderItemModel();
+            m_orderItem = i_item;
+            OrderStatus = i_item.m_State;
+            OderID = i_item.m_OrdrID;
+            OrderItemName = i_item.m_OrderItemName;
+            OrderItemDescription = i_item.m_Description;
+            FirstTimeToShowString = OrderItemTimeCreate;
+
         }
 
-        private void CreateGrid()
+        private void feelColorsState()
         {
-            m_Grid = new Grid
-            {
-                Margin = new Thickness(0, 5, 0, 5),
-                IsVisible = false,
-                RowDefinitions =
-                {
-                    new RowDefinition {Height = new GridLength(2, GridUnitType.Star)},
-                    new RowDefinition {Height = new GridLength(2, GridUnitType.Star)},
-                    new RowDefinition {Height = new GridLength(2, GridUnitType.Star)},
-                    new RowDefinition {Height = new GridLength(2, GridUnitType.Star)},
-                    new RowDefinition {Height = new GridLength(2, GridUnitType.Star)},
-                    new RowDefinition {Height = new GridLength(2, GridUnitType.Star)}
-
-                },
-                ColumnDefinitions =
-                {
-                    new ColumnDefinition() {Width = new GridLength(2, GridUnitType.Star)},
-                    new ColumnDefinition() {Width = new GridLength(2, GridUnitType.Star)},
-                    new ColumnDefinition() {Width = new GridLength(2, GridUnitType.Star)}
-                },
-                BackgroundColor = Color.CadetBlue
-            };
+           m_StateColors.Add(eOrderItemState.Waiting, Color.Firebrick);
+           m_StateColors.Add(eOrderItemState.Available, Color.Gold);
+           m_StateColors.Add(eOrderItemState.InPreparation, Color.DarkOrange);
+           m_StateColors.Add(eOrderItemState.Ready, Color.DarkGreen);
+           m_StateColors.Add(eOrderItemState.Completed, Color.YellowGreen);
```

Comment:
Since these state colors are added to each and every OrderItemView, is it not better to define m_StateColors as static and initialize it in the static way?

Reply:
@AvivBk  You need to change this also, and move initialization to static init

Reply:
Also as we discussed, this should be in the static initializer for m_StateColors.